src/solver/simulation/include/antares/solver/simulation/solver.hxx

```diff
@@ -695,74 +722,86 @@ void ISimulation<ImplementationType>::computeRandomNumbers(randomNumbers& random
-            // looking for the initial reservoir level (begining of the year)
-            auto& min = area.hydro.reservoirLevel[Data::PartHydro::minimum];
-            auto& avg = area.hydro.reservoirLevel[Data::PartHydro::average];
-            auto& max = area.hydro.reservoirLevel[Data::PartHydro::maximum];
-
-            // Month the reservoir level is initialized according to.
-            // This month number is given in the civil calendar, from january to december (0 is
-            // january).
-            int initResLevelOnMonth = area.hydro.initializeReservoirLevelDate;
-
-            // Conversion of the previous month into simulation calendar
-            int initResLevelOnSimMonth = study.calendar.mapping.months[initResLevelOnMonth];
-
-            // Previous month's first day in the year
-            int firstDayOfMonth = study.calendar.months[initResLevelOnSimMonth].daysYear.first;
-
-            double randomLevel = randomReservoirLevel(min[firstDayOfMonth],
-                                                                       avg[firstDayOfMonth],
-                                                                       max[firstDayOfMonth],
-                                                                       randomHydroGenerator);
-
-            // Possibly update the intial level from scenario builder
-            if (study.parameters.useCustomScenario)
-            {
-                double levelFromScenarioBuilder = study.scenarioHydroLevels[areaIndex][y];
-                if (levelFromScenarioBuilder >= 0.)
-                    randomLevel = levelFromScenarioBuilder;
-            }
-
-            if (pHydroHotStart)
-            {
-                if (!isPerformed || !area.hydro.reservoirManagement)
-                {
-                    // This initial level should be unused, so -1, as impossible value, is suitable.
-                    randomForYears.pYears[indexYear].pReservoirLevels[areaIndex] = -1.;
-                    areaIndex++;
-                    return; // Skipping the current area
-                }
-
-                if (!pFirstSetParallelWithAPerformedYearWasRun)
-                    randomForYears.pYears[indexYear].pReservoirLevels[areaIndex] = randomLevel;
-                // Else : means the start levels (multiple areas are affected) of a year are
-                // retrieved from a previous year and
-                //		  these levels are updated inside the year job (see year job).
-            }
-            else
-            {
-                // Current area's hydro starting (or initial) level computation
-                // (no matter if the year is performed or not, we always draw a random initial
-                // reservoir level to ensure the same results)
-                if (isPerformed)
-                    randomForYears.pYears[indexYear].pReservoirLevels[areaIndex] = randomLevel;
-            }
-
-            areaIndex++;
-        }); // each area
+        study.areas.each(
```

Comment:
lambda first part rejected next line : is that what we want ?  Especially if it's not indented in comparison to the previous line

Reply:
This is because otherwise the function body will be too far right, I don't think we can do much about this.

A possibility is to use [OuterScope](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#lambdabodyindentation) instead of signature for lambda body indentation, but it will lead to weird formatting in other cases, where the lambda body will be on the left, but argument parameters on the right.